### PR TITLE
deprecating the need for interp_u method in integrators

### PR DESCRIPTION
**Pull request by drgona**
Propose simplification of ODE integrators in integrators.py

### CODE REVIEW

**Comment by drgona** [comment]
Before merge all non-autonomous ODE system ID and control examples needs to be updated with removed interp_u

**Comment by madelynshapiro** [comment]
@drgona I recommend marking this PR as a !Draft to block merging until all remaining tasks are complete

File: src/neuromancer/dynamics/integrators.py
Comment (madelynshapiro) [inline src/neuromancer/dynamics/integrators.py:28]:
```suggestion
            warnings.warn('interp_u argument is deprecated, it has no effect, and will be removed in the next release.', FutureWarning)
```

https://docs.python.org/3/library/exceptions.html#FutureWarning

File: src/neuromancer/dynamics/integrators.py
Comment (madelynshapiro) [inline src/neuromancer/dynamics/integrators.py:80]: Is there any case where the user might want to set a non-zero `t_0`?

File: src/neuromancer/dynamics/integrators.py
Comment (madelynshapiro) [inline src/neuromancer/dynamics/integrators.py:154]: The old code modifies `t` to pass the `tq` arg to `self.state` for `k2` - `k4`. Does this need to be handled in the new implementation or is `tq` functionality being deprecated?

Comment (madelynshapiro) [reply src/neuromancer/dynamics/integrators.py:154]: If deprecating, update function def inline comments to remove `t_i` term

Comment (madelynshapiro) [reply src/neuromancer/dynamics/integrators.py:154]: Same comment applies to all other classes' `integrate` methods implemented below

**Comment by madelynshapiro** [comment]
@drgona Almost ready to merge:

- [x] Tests passing
- [ ] Resolve minor review comments above
- [ ] Update version number to 1.4.1 in `pyproject.toml`
- [ ] Summarize changes for 1.4.1 in main `README.md`

**Review by madelynshapiro — APPROVED** [review]

**Pull request merged**
